session.lisp

```diff
@@ -297,6 +297,25 @@ will not create a new one."
       (setq session nil))
     session))
 
+(defgeneric session-regenerate-cookie-value (session)
+  (:documentation "Regenerates the cookie value. This should be used
+when a user logs in according to the application to prevent against
+session fixation attacks."))
+
+(defmethod session-regenerate-cookie-value ((session session))
+  "The cookie value being dependent on ID, USER-AGENT, REMOTE-ADDR,
+START, and *SESSION-SECRET*, the only value we can change is START
+to regenerate a new value. Since we're generating a new cookie, it
+makes sense to have the session being restarted, in time. That said,
+because of this fact, calling this function twice in the same second
+will regenerate twice the same value."
```

Comment:
- The docstring and the documentation should have precisely the same wording.
- I don't think that a generic function is useful at this point.  It might be useful to have the session string encoding functionality be a generic function, but that require refactoring of `STRINGIFY-SESSION` and `ENCODE-SESSION-STRING` which I think would be a lot of work with uncertain benefit.